src/rating/rating.js

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+
+// @flow
+
+import React from 'react';
+import {KIND} from './constants';
+import type {RatingPropsT, RatingStateT} from './types';
+import {StyledRoot, StyledStar, StyledEmoticon} from './styled-components';
+import {getOverrides} from '../helpers/overrides';
+
+const ENTER_KEY_CODE = 13;
+const SPACE_KEY_CODE = 32;
+
+class Rating extends React.Component<RatingPropsT, RatingStateT> {
+  static defaultProps = {
+    overrides: {},
+    kind: KIND.star,
+  };
+
+  state = {};
+
+  selectItem = (value: number) => {
+    const {onChange} = this.props;
+
+    onChange && onChange({value});
+    this.setState({previewIndex: undefined});
+  };
+
+  updatePreview = (previewIndex?: number) => {
+    this.setState({previewIndex});
+  };
+
+  renderRatingContents = () => {
+    const {overrides = {}, value = -1} = this.props;
+    const {previewIndex} = this.state;
+
+    const [Star, starProps] = getOverrides(overrides.Star, StyledStar);
+    const [Emoticon, emoticonProps] = getOverrides(
+      overrides.Emoticon,
+      StyledEmoticon,
+    );
+
+    const ratings = [];
+    const RatingItem = this.props.kind === KIND.star ? Star : Emoticon;
+    const overrideProps =
+      this.props.kind === KIND.star ? starProps : emoticonProps;
+
+    for (let x = 1; x <= 5; x++) {
```

Comment:
Should we make the `5` be overridable through props?

Reply:
and also need to provide new set of images

Reply:
I originally thought about that, but it would complicate the component for a feature that we aren't sure exists or is needed.  If there is a Figma spec for a dynamic rating than we can probably add that functionality in at a later date.

For the Star component it's trivial, but the Emoticon example is limited to 5 items.

Reply:
so I suggest to talk about it with @nguyensomniac or Jeff. As our customers likely to request this feature. For instance, currently some uber apps have 2 or 3 only rating icons.

Reply:
That would also fix some of the complexity in logic between the 2 IMHO

Reply:
These components are now split. Much cleaner!

Reply:
Agreed with @austin94 on this, let's wait until we know if this is a feature people need.